tests/test_document.py

```diff
@@ -127,70 +138,45 @@ def test_get_by_uri(self):
         assert doc
         assert_equal(doc['title'], "document1") 
 
-    def test_get_all_by_uri(self):
+    @staticmethod
+    def test_get_all_by_uri():
```

Comment:
Really just the test_save methods you have would be fine, as long as you have maybe:

- The simple case of saving a new doc
- The case of merging into an old doc
- The case of two, disjoint docs being merged by a third
- Bonus: test that iteration works

Otherwise, these are testing the implementation rather than the public behavior, which means that if the implementation details change the tests have to change and therefore it's harder to know if the tests are broken or the code is broken.